_posts/2019-02-19-SageMaker-In-Action.md

```diff
@@ -0,0 +1,177 @@
+---
+layout: post
+title: AWS SageMaker In Action
+banner: /assets/posts/2019-02-19-SageMaker-In-Action/banner.jpg
+author: mkianpour
+
+---
+In the [previous post](https://www.sentialabs.io/2019/01/30/SageMaker-In-Action.html) we collected and prepared training data to build the model. Now we continue with the building and deploying the model:
+
+### Step 2-2: Training and Building the Model
+
+To build the model we should have training code. Amazon SageMaker is quite flexible in using different algorithms. It also offers some ready to use algorithms. From [here](https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms.html):
+
+>Amazon SageMaker algorithms are packaged as Docker images. This gives you the flexibility to use almost any algorithm code with Amazon SageMaker, regardless of implementation language, dependent libraries, frameworks, and so on.
+
+Here we use an available algorithm as mentioned: [BlazingText](https://docs.aws.amazon.com/sagemaker/latest/dg/blazingtext.html). We brought a Notebook up and use the algorithm to train the model. Bringing up a Jupyter Notebook is almost straight forward, as usual specify instance type and some AWS specific information such as VPC, encryption and most importantly IAM Role. When we have our Notebook running, we can run our code. You can see the whole notebook in [Github repository](https://github.com/mkianpour/twitter-machine-learning). The relevant code for training data is as follows:
+
+```python
+import sagemaker
+from sagemaker import get_execution_role
+import json
+import boto3
+import pandas as pd
+
+sess = sagemaker.Session()
+
+role = get_execution_role()
+print(role) # This is the role that SageMaker would use to leverage AWS resources (S3, CloudWatch) on your behalf
+
+bucket = "XYZ" # Replace with your own bucket name if needed
```

Comment:
Use an ENV var instead?

Reply:
it's proof of concept. you can create Notebooks with CloudFormation but here not used and ENV var is not meaningful.